keyboards/duck/orion/v3/matrix.c

```diff
@@ -0,0 +1,272 @@
+/*
+Copyright 2019 MechMerlin <mechmerlin@gmail.com>
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <util/delay.h>
+#include <avr/io.h>
+#include <stdio.h>
+#include "matrix.h"
+#include "util.h"
+#include "print.h"
+#include "debug.h"
+
+static uint8_t debouncing = DEBOUNCE;
+
+/* matrix state(1:on, 0:off) */
+static matrix_row_t matrix[MATRIX_ROWS];
+static matrix_row_t matrix_debouncing[MATRIX_ROWS];
+
+static uint8_t read_rows(uint8_t col);
+static void init_rows(void);
+static void unselect_cols(void);
+static void select_col(uint8_t col);
+
+
+__attribute__ ((weak))
+void matrix_init_kb(void) {
+    matrix_init_user();
+}
+
+__attribute__ ((weak))
+void matrix_scan_kb(void) {
+    matrix_scan_user();
+}
+
+__attribute__ ((weak))
+void matrix_init_user(void) {
+}
+
+__attribute__ ((weak))
+void matrix_scan_user(void) {
+}
+
+void backlight_init_ports(void)
+{
+  DDRD  |=  0b11010000;
+  PORTD &= ~0b01010000;
+  PORTD |=  0b10000000;
+  DDRB  |=  0b00011111;
+  PORTB &= ~0b00001110;
+  PORTB |=  0b00010001;
+  DDRE  |=  0b01000000;
+  PORTE &= ~0b01000000;
+}
+
+void matrix_init(void) {
+  backlight_init_ports();
+  unselect_cols();
+  init_rows();
+
+  for (uint8_t i=0; i < MATRIX_ROWS; i++)  {
+    matrix[i] = 0;
+    matrix_debouncing[i] = 0;
+  }
+
+  matrix_init_quantum();
+}
+
+uint8_t matrix_scan(void) {
```

Comment:
Could this be updated to use the debounce API?

Reply:
Are you talking about the eager stuff? Can you point me to an example of it being used and I'll see if it can be implemented with line decoders.

Reply:
Asid from: 
https://github.com/qmk/qmk_firmware/blob/41b9be560d68a69a0c87d90e94700082e1cbbd3a/quantum/matrix.c#L259-L278

No, not really

Reply:
Ahh I remember this conversation. The answer is no it did not work with line decoders, or I couldn't get it to work. This will require more more indepth digging.